### PR TITLE
feat(light-client): complete light client state

### PR DESCRIPTION
**Pull request by mrain**
closes: #924 , #929 

This PR adds fee ledger commitment and block tree commitment to the light client state.

### CODE REVIEW

**Review by jbearer — APPROVED** [review]

File: sequencer/src/state_signature.rs [hunk 1]
```diff
@@ -38,47 +42,73 @@ pub(super) async fn state_signature_loop<N>(
...
     stake_table_comm: &StakeTableCommitmentType,
-) -> LightClientState {
-    // TODO(Chengyu): fill these `default()` with actual value
-    LightClientState {
+) -> Result<LightClientState, PrimitivesError> {
+    let header = leaf.get_block_header();
+    let mut block_comm_root_bytes = vec![];
+    header
+        .block_merkle_tree_root
+        .serialize_compressed(&mut block_comm_root_bytes)?;
```
Comment (alxiong) [inline sequencer/src/state_signature.rs:99]: why is the block header being the block commitment root?
we are supposed to build a merkle tree out of all the block commitment, right?

File: sequencer/src/state_signature.rs [hunk 2]
```diff
@@ -38,47 +42,73 @@ pub(super) async fn state_signature_loop<N>(
...
 
+fn hash_bytes_to_field(bytes: &[u8]) -> Result<CircuitField, PrimitivesError> {
+    // make sure that `mod_order` won't happen.
+    let bytes_len = ((<CircuitField as PrimeField>::MODULUS_BIT_SIZE + 7) / 8 - 1) as usize;
+    let elem = bytes
+        .chunks(bytes_len)
+        .map(CircuitField::from_le_bytes_mod_order)
+        .collect::<Vec<_>>();
+    Ok(VariableLengthRescueCRHF::<_, 1>::evaluate(elem)?[0])
+}
```
Comment (alxiong) [inline sequencer/src/state_signature.rs:89]: are there reasons to use Rescue Hash here? why can't we just hash down to 32 bytes, and then use `jf_utils::bytes_to_field_elements`?

File: sequencer/src/state_signature.rs (hunk 1 above)
Comment (mrain) [reply sequencer/src/state_signature.rs:99]: Yes and this is the Merkle root. Just naming

File: sequencer/src/state_signature.rs (hunk 2 above)
Comment (mrain) [reply sequencer/src/state_signature.rs:89]: No specific reason, We could use others as well

Comment (alxiong) [reply sequencer/src/state_signature.rs:89]: yeah, let's try to minimize rescue usage wheneve possible, it's slow and not as time-tested. 
also try to reuse jf_utils, cuz converting bytes to fields is a common util. 
would you agree?

File: sequencer/src/state_signature.rs (hunk 1 above)
Comment (alxiong) [reply sequencer/src/state_signature.rs:99]: i think the BlockHeader is [more than just block_commit_root](https://github.com/EspressoSystems/espresso-sequencer/blob/main/sequencer/src/header.rs#L79), right? even though I don't see a problem, but it's not technically this bytes is committing to an superset of system states than just block_comm_root.

**Pull request merged**